### PR TITLE
[NCL-7256] Keep UI partially working even if KeyCloak is not available

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

**Comment by matedo1** [comment]
Hi @jomrazek , could you please do a review for @patrikk0123 ?

**Review by jomrazek — COMMENTED** [review]
looks ok to me

File: src/AppRoutes.tsx [hunk 1]
```diff
@@ -18,20 +18,25 @@ import { ProjectDetailPage } from './components/ProjectDetailPage/ProjectDetailP
 // special pages
 import { DemoPage } from './components/DemoPage/DemoPage';
 import { VariablesPage } from './components/VariablesPage/VariablesPage';
-import { NotFoundPage } from './components/NotFoundPage/NotFoundPage';
 import { AdministrationPage } from './components/AdministrationPage/AdministrationPage';
 import { keycloakService, AUTH_ROLE } from './services/keycloakService';
+import { ErrorPage } from './components/ErrorPage/ErrorPage';
 
 interface IProtectedRouteProps {
+  title?: string;
```
Comment (matedo1) [inline src/AppRoutes.tsx:26]: I would be strict in this case for now, let's remove `?` character

File: src/AppRoutes.tsx [hunk 2]
```diff
@@ -81,12 +86,12 @@ export const AppRoutes = () => (
...
         element={
-          <ProtectedRoute role={AUTH_ROLE.Admin}>
+          <ProtectedRoute title="Administration" role={AUTH_ROLE.Admin}>
             <AdministrationPage />
           </ProtectedRoute>
         }
       />
     </Route>
-    <Route path="*" element={<NotFoundPage />} />
+    <Route path="*" element={<ErrorPage errorDescription="The requested resource could not be found."></ErrorPage>} />
```
Comment (matedo1) [inline src/AppRoutes.tsx:95]: I would use also `title` attribute `Page not found`, but define it inside https://github.com/project-ncl/pnc-web-ui-react/blob/main/src/utils/PageTitles.ts like `pageNotFound`

File: src/components/ErrorPage/ErrorPage.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { CubesIcon } from '@patternfly/react-icons';
+import { Link } from 'react-router-dom';
+import { StateCard } from '../EmptyStates/StateCard';
+import { PageLayout } from '../PageLayout/PageLayout';
+import styles from './ErrorPage.module.css';
+
+interface IErrorPage {
+  pageTitle?: string;
```
Comment (matedo1) [inline src/components/ErrorPage/ErrorPage.tsx:8]: I would be strict also in this case, let's remove `?`, then also default values bellow will be removed

File: src/components/ErrorPage/ErrorPage.module.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.return-button-margin {
+  margin-top: 15px;
```
Comment (matedo1) [inline src/components/ErrorPage/ErrorPage.module.css:2]: for generic margins, paddings, etc, we use so called generics (see index.css), let's use `m-t-15: 15px` and remove this css module

File: src/AppRoutes.tsx [hunk 5]
```diff
@@ -52,15 +57,15 @@ export const AppRoutes = () => (
       <Route
         path="create"
         element={
-          <ProtectedRoute>
+          <ProtectedRoute title="Create Project">
```
Comment (matedo1) [inline src/AppRoutes.tsx:60]: let's define all titles inside https://github.com/project-ncl/pnc-web-ui-react/blob/main/src/utils/PageTitles.ts

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/services/pncClient.ts [hunk 6]
```diff
@@ -26,7 +26,7 @@ class PncClient {
     httpClient.interceptors.request.use((config) => {
       // perform actions before request is sent
 
-      if (keycloakService.isAuthenticated()) {
+      if (keycloakService.isKeycloakAvailable && keycloakService.isAuthenticated()) {
```
Comment (jomrazek) [inline src/services/pncClient.ts:29]: if we care just about true or false, can't we move isKeycloakAvailable check into service? Same for getUser and hasRealmRole in appLayout? Seems like we would need to do this check in all classes when using those methods while they can return false from service when we don't need to differentiate the failure point.

**Comment by patrikk0123** [comment]
Changes had to be done, continuation here:
https://github.com/project-ncl/pnc-web-ui-react/pull/87

**Pull request closed without merging**